support-frontend/assets/components/ticker/contributionTicker.jsx

```diff
@@ -61,56 +68,77 @@ export default class ContributionTicker extends Component<PropTypes, StateTypes>
 
       if (totalSoFar >= goal) {
         this.onGoalReached();
+        this.classModifiers.push('goal-reached');
+        this.goalReached = true;
       }
 
       this.setState({ totalSoFar: initialTotal, goal });
-      window.setTimeout(() => {
-        window.requestAnimationFrame(this.increaseCounter);
-        this.animateBar(totalSoFar, goal);
-      }, 500);
+      window.requestAnimationFrame(this.increaseTextCounter);
     });
   }
 
   onGoalReached: () => void;
   tickerJsonUrl: string;
   totalSoFar: number;
   count = 0;
-  filledProgressBar: ?HTMLDivElement;
-  increaseCounter = this.increaseCounter.bind(this);
-
-  animateBar(totalSoFar: number, goal: number) {
-    const progressBarElement = this.filledProgressBar;
-    if (progressBarElement && progressBarElement instanceof HTMLElement) {
-      progressBarElement.style.transform = `translateX(${percentageTotalAsNegative(totalSoFar, goal)}%)`;
-    }
-  }
+  tickerType: TickerType;
+  classModifiers: Array<?string>;
+  goalReached: boolean;
 
-  increaseCounter() {
+  increaseTextCounter = () => {
     const { totalSoFar } = this;
     this.count += Math.floor(totalSoFar / 100);
 
     if (this.count >= this.totalSoFar) {
       this.setState({ totalSoFar: this.totalSoFar });
     } else {
       this.setState({ totalSoFar: this.count });
-      window.requestAnimationFrame(this.increaseCounter);
+      window.requestAnimationFrame(this.increaseTextCounter);
     }
   }
 
+  renderContributedSoFar = () => {
+    if (!this.goalReached) {
```

Comment:
you could structure this as follows, if you find it clear. slightly more concise but not much in it.

```javascript

if (this.goalReached) {
  if (this.tickerType === 'unlimited') {
    ...
  } else if (this.tickerType === 'hardstop') {
    ...
  }
}

return (
  <div className="contributions-landing-ticker__so-far">
    <div className="contributions-landing-ticker__count">${Math.floor(this.state.totalSoFar).toLocaleString()}</div>
    <div className="contributions-landing-ticker__count-label contributions-landing-ticker__label">contributed</div>
  </div>
);
```

Reply:
For me, it's easier to read/understand a longer list of parameters rather than looping inside a loop

Reply:
👍 for the clarity of this function